src/lib/Libtpp/tpp_em.c

```diff
@@ -1090,7 +1090,10 @@ tpp_mbox_read(tpp_mbox_t *mbox, unsigned int *tfd, int *cmdval, void **data)
 	if (cmd == NULL) {
 		mbox->mbox_size = 0;
 #ifdef HAVE_SYS_EVENTFD_H
-		read(mbox->mbox_eventfd, &u, sizeof(uint64_t));
+		if (read(mbox->mbox_eventfd, &u, sizeof(uint64_t)) == -1) {
+			tpp_log(LOG_CRIT, __func__, "Unable to read from msg box");
+			return -1;
```

Comment:
```suggestion
	                tpp_unlock(&mbox->mbox_mutex);
			return -1;
```

There may be a deadlock if this is not done